build.gradle

```diff
@@ -0,0 +1,142 @@
+plugins {
+    id 'checkstyle' apply true
+    id 'io.spring.dependency-management' version '1.0.3.RELEASE'
+    id 'org.springframework.boot' version '1.5.9.RELEASE' apply true
+    id 'com.jfrog.bintray' version '1.8.0' apply true
+    id 'java-library' apply true
+    id 'maven' apply true
+    id 'maven-publish' apply true
+}
+
+group 'uk.gov.hmcts.reform'
+version '1.0.0'
+
+checkstyle {
+    maxWarnings = 0
+    toolVersion = '8.3'
+    // need to set configDir to rootDir otherwise submodule will use submodule/config/checkstyle
+//    configDir = new File(rootDir, 'config/checkstyle')
+}
+
+sourceCompatibility = 1.8
+
+compileJava {
+    options.compilerArgs << '-parameters' << '-Xlint:deprecation'
+}
+
+compileTestJava {
+    options.compilerArgs << '-Xlint:deprecation'
+}
+
+repositories {
+    maven {
+        url "https://dl.bintray.com/hmcts/hmcts-maven"
+    }
+    jcenter()
+    mavenLocal()
+}
+
+jar {
+    manifest {
+        attributes 'Implementation-Title': project.name, 'Implementation-Version': project.version
+    }
+}
+
+task printVersion {
+    doLast {
+        print project.version
+    }
+}
+
+def pomConfig = {
+    licenses {
+        license {
+            name "MIT License"
+            url "http://www.opensource.org/licenses/mit-license.php"
+            distribution "repo"
+        }
+    }
+
+    scm {
+        url "https://github.com/hmcts/service-authorisation-access-client"
+    }
+}
+
+
+task sourcesJar(type: Jar, dependsOn: classes) {
+    classifier = 'sources'
+    from sourceSets.main.allSource
+}
+
+task javadocJar(type: Jar, dependsOn: javadoc) {
+    classifier = 'javadoc'
+    from javadoc.destinationDir
+}
+
+artifacts {
+    archives sourcesJar
+    archives javadocJar
+}
+
+publishing {
+    publications {
+        Main(MavenPublication) {
+            from components.java
+            artifact sourcesJar
+            artifact javadocJar
+            groupId project.group
+            artifactId 'service-authorisation-access-client'
+            version project.version
+
+            pom.withXml {
+                def root = asNode()
+                root.appendNode('description', 'Client library for communicating with idam service authorisation access')
+                root.appendNode('name', 'Service authorisation access client')
```

Comment:
needs an update on this line, the one above and the one below

Reply:
done